test/uk/gov/hmrc/helptosavefrontend/models/NSIUserInfoSpec.scala

```diff
@@ -335,5 +328,37 @@ class NSIUserInfoSpec extends WordSpec with Matchers with GeneratorDrivenPropert
       }
 
     }
+
+    "transform" should {
+
+      "remove new line, tab and carriage return in forename" in {
+        val modifiedForename = "\n\t\rname\t"
+        val userInfo = NSIUserInfo(validUserInfo.copy(forename = modifiedForename))
+        (userInfo.map{_.forename} == Valid("name"))
+      }
+
+      "remove white spaces in forename" in {
+        val forenameWithSpaces = " " + "forename" + " "
+        val userInfo = NSIUserInfo(validUserInfo.copy(forename = forenameWithSpaces))
+        (userInfo.map{_.forename} == Valid("forename"))
+      }
+
+      "remove whitespace from surname" in {
+        val userInfo = NSIUserInfo(validUserInfo.copy(surname = " " + surname))
+        (userInfo.map{_.surname} == Valid(" " + surname)) shouldBe false
```

Comment:
should probably check that we get `surname` back rather than asserting that we don't get `" " + surname` - it could not be `" " + surname` and not be equal to `surname`